### PR TITLE
ISPN-12726 Trust Store

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-12726

### CODE REVIEW

File: server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml [hunk 1]
```diff
@@ -73,21 +76,22 @@
...
                   <ssl>
                      <keystore path="ServerConfigurationParserTest-keystore.pfx"
                                relative-to="infinispan.server.config.path"
-                               alias="server">
+                               alias="server" provider="SunJSSE">
                         <credential-reference store="credentials" alias="keystore"/>
                      </keystore>
+                     <truststore path="truststore.p12" provider="SunJSSE"
+                                 relative-to="infinispan.server.config.path">
+                        <credential-reference store="credentials" alias="trust"/>
```
Comment (ryanemerson) [inline server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml:93]: <credential-reference> usage doesn't appear to be documented anywhere, neither for this use-case or it's use in general.

File: server/runtime/src/main/java/org/infinispan/server/configuration/security/TrustStoreConfigurationBuilder.java [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+import org.infinispan.commons.CacheConfigurationException;
+import org.infinispan.commons.configuration.Builder;
+import org.infinispan.commons.configuration.attributes.AttributeSet;
+import org.infinispan.configuration.parsing.ParseUtils;
+import org.infinispan.server.Server;
+import org.wildfly.security.keystore.KeyStoreUtil;
+
+/**
+ * @since 10.0
+ */
```
Comment (ryanemerson) [inline server/runtime/src/main/java/org/infinispan/server/configuration/security/TrustStoreConfigurationBuilder.java:25]: Outdated :slightly_smiling_face:

File: server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationParser.java [hunk 3]
```diff
@@ -1008,7 +1018,75 @@ private void parseKeyStore(XMLExtendedStreamReader reader, ServerConfigurationBu
...
+         throw Server.log.missingCredential(Element.TRUSTSTORE.toString(), Attribute.PASSWORD.toString());
+      }
+      if (element != null) {
+         throw ParseUtils.unexpectedElement(reader, element);
+      }
+      trustStoreBuilder.build();
+   }
+
+   private void parseTrustStoreRealm(XMLExtendedStreamReader reader, ServerConfigurationBuilder builder, TrustStoreRealmConfigurationBuilder trustStoreBuilder, ServerIdentitiesConfigurationBuilder serverIdentitiesConfigurationBuilder) throws XMLStreamException {
+      String name = "trust";
```
Comment (ryanemerson) [inline server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationParser.java:1071]: Should this just be the default value for the attribute in `TrustStoreRealmConfiguration`?

Comment (tristantarrant) [reply server/runtime/src/main/java/org/infinispan/server/configuration/ServerConfigurationParser.java:1071]: Probably, but this affects all realms, so let's postpone this clean up for another PR

File: documentation/src/main/asciidoc/topics/config_examples/server_ssl_identity_truststore.xml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+         <server-identities>
+            <ssl>
+              <!-- Keystores contain certificate chains that verify server identities to clients. -->
+              <!-- Keystores are relative to the server/conf directory by default. -->
+               <keystore path="server.p12"
+                         relative-to="infinispan.server.config.path"
+                         keystore-password="secret"
+                         alias="server"/>
+               <!-- Trust stores hold client certificates. -->
+               <!-- If you add a trust store, clients must present certificates when connecting to the server. -->
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply documentation/src/main/asciidoc/topics/config_examples/server_ssl_identity_truststore.xml:16]: They are still valid for explicit security configs.

File: server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml (hunk 1 above)
Comment (tristantarrant) [reply server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml:93]: https://github.com/infinispan/infinispan/pull/9070/files#diff-6e880072938cdbf4757fce0f304a80a85daf06c40eedcf2b8c83aa3563869b5fR15

**Comment by tristantarrant** [comment]
@oraNod you may want to check this

**Comment by oraNod** [comment]
> @oraNod you may want to check this

@tristantarrant Hopefully not missing something obvious but I pushed a commit with some suggested doc edits this morning. Did you revert that or did it get wiped out on the last push?

**Comment by tristantarrant** [comment]
I must have repushed it... sorry

File: server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml (hunk 1 above)
Comment (ryanemerson) [reply server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml:93]: I meant the of credential-reference. This isn't an issue just with this PR, more of an omission from the docs in general.

Comment (oraNod) [reply server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml:93]: @ryanemerson We do, in fact, document credential-reference. Please see here: https://infinispan.org/docs/stable/titles/server/server.html#server_credentials

Comment (ryanemerson) [reply server/runtime/src/test/resources/configuration/ServerConfigurationParserTest.xml:93]: Sorry I obviously missed that. Intellij didn't find anything when I hunted for `<credential-reference`.

**Pull request merged**